### PR TITLE
funcx ws draft

### PR DESCRIPTION
**Pull request by Loonride**

### CODE REVIEW

File: funcx_forwarder/forwarder.py [hunk 1]
```diff
@@ -354,6 +357,18 @@ def handle_results(self):
                 task.exception = message['exception']
                 task.completion_time = time.time()
 
+            batch_id = task.batch_id
+            if 'result' in message or 'exception' in message and batch_id:
+                connection = pika.BlockingConnection(self.rabbitmq_conn_params)
```
Comment (yadudoc) [inline funcx_forwarder/forwarder.py:362]: This is okay for debugging, but we should avoid opening and closing connections for each task. We could do something like `self.publish_task_update(task_id, taskgroup_id)` for each task.

**Review by yadudoc — COMMENTED** [review]
Looks reasonable. We might need some minor refactoring, perhaps after some discussion.

File: funcx_forwarder/forwarder.py (hunk 1 above)
Comment (yadudoc) [reply funcx_forwarder/forwarder.py:362]: At this point, I'd add some log lines to better capture time spent in opening the connection and publishing the message.
We can defer optimization until we have a better picture of how expensive this operation is.

**Review by ryanchard — APPROVED** [review]
Looks good!

**Pull request merged**